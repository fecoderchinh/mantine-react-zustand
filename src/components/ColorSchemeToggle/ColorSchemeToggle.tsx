import { Button, Group, useMantineColorScheme } from '@mantine/core'

export const ColorSchemeToggle = () => {
   const { setColorScheme } = useMantineColorScheme()

   return (
      <Group justify='center' mt='xl'>
         <Button onClick={() => setColorScheme('light')}>Light</Button>
         <Button onClick={() => setColorScheme('dark')}>Dark</Button>
         <Button onClick={() => setColorScheme('auto')}>Auto</Button>
      </Group>
   )
}

export default ColorSchemeToggle
