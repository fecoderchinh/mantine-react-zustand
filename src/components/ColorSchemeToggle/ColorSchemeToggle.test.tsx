import { render } from '@tests'
import { ColorSchemeToggle } from './ColorSchemeToggle'

describe('ColorSchemeToggle', () => {
   it('should render the ColorSchemeToggle', () => {
      const { container } = render(<ColorSchemeToggle />)
      expect(container).toMatchSnapshot()
   })
})
