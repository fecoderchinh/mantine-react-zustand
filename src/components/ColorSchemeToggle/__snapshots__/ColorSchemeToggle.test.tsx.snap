// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSchemeToggle should render the ColorSchemeToggle 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: center; --group-wrap: wrap;"
  >
    <button
      class="mantine-focus-auto mantine-active m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
      type="button"
    >
      <span
        class="m-80f1301b mantine-Button-inner"
      >
        <span
          class="m-811560b9 mantine-Button-label"
        >
          Light
        </span>
      </span>
    </button>
    <button
      class="mantine-focus-auto mantine-active m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
      type="button"
    >
      <span
        class="m-80f1301b mantine-Button-inner"
      >
        <span
          class="m-811560b9 mantine-Button-label"
        >
          Dark
        </span>
      </span>
    </button>
    <button
      class="mantine-focus-auto mantine-active m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
      type="button"
    >
      <span
        class="m-80f1301b mantine-Button-inner"
      >
        <span
          class="m-811560b9 mantine-Button-label"
        >
          Auto
        </span>
      </span>
    </button>
  </div>
</div>
`;
